spec/rpc/README.md

```diff
@@ -0,0 +1,1264 @@
+- `page (integer)`:
+- `per_page (integer)`:
+
+#### Request
+
+##### HTTP
+
+```sh
+curl  http://127.0.0.1:26657/validators
+```
+
+##### JSONRPC
+
+```sh
+curl -X POST https://localhost:26657 -d "{\"jsonrpc\":\"2.0\",\"id\":1,\"method\":\"validators\",\"params\":{\"height\":\"1\", \"page\":\"1\", \"per_page\":\"20\"}}"
+```
+
+#### Response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 0,
+  "result": {
+    "block_height": "55",
+    "validators": [
+      {
+        "address": "000001E443FD237E4B616E2FA69DF4EE3D49A94F",
+        "pub_key": {
+          "type": "tendermint/PubKeyEd25519",
+          "value": "9tK9IT+FPdf2qm+5c2qaxi10sWP+3erWTKgftn2PaQM="
+        },
+        "voting_power": "239727",
+        "proposer_priority": "-11896414"
+      }
+    ],
+    "count": "1",
+    "total": "25"
+  }
+}
+```
+
+### Genesis
+
+Get Genesis of the chain.
+
+#### Request
+
+##### HTTP
+
+```sh
+curl  http://127.0.0.1:26657/genesis
+```
+
+##### JSONRPC
+
+```sh
+curl -X POST https://localhost:26657 -d "{\"jsonrpc\":\"2.0\",\"id\":1,\"method\":\"genesis\"}"
+```
+
+#### Response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 0,
+  "result": {
+    "genesis": {
+      "genesis_time": "2019-04-22T17:00:00Z",
+      "chain_id": "cosmoshub-2",
+      "initial_height": "2",
+      "consensus_params": {
+        "block": {
+          "max_bytes": "22020096",
+          "max_gas": "1000",
+          "time_iota_ms": "1000"
+        },
+        "evidence": {
+          "max_age": "100000"
+        },
+        "validator": {
+          "pub_key_types": [
+            "ed25519"
+          ]
+        }
+      },
+      "validators": [
+        {
+          "address": "B00A6323737F321EB0B8D59C6FD497A14B60938A",
+          "pub_key": {
+            "type": "tendermint/PubKeyEd25519",
+            "value": "cOQZvh/h9ZioSeUMZB/1Vy1Xo5x2sjrVjlE/qHnYifM="
+          },
+          "power": "9328525",
+          "name": "Certus One"
+        }
+      ],
+      "app_hash": "",
+      "app_state": {}
+    }
+  }
+}
+```
+
+### ConsensusParams
+
+Get the consensus parameters.
+
+#### Request
+
+##### HTTP
+
+```sh
+curl  http://127.0.0.1:26657/consensus_params
+```
+
+##### JSONRPC
+
+```sh
+curl -X POST https://localhost:26657 -d "{\"jsonrpc\":\"2.0\",\"id\":1,\"method\":\"consensus_params\"}"
+```
+
+#### Response
+
+```json
+{
+  "jsonrpc": "2.0",
+  "id": 0,
+  "result": {
+    "block_height": "1",
+    "consensus_params": {
+      "block": {
+        "max_bytes": "22020096",
+        "max_gas": "1000",
+        "time_iota_ms": "1000"
+      },
+      "evidence": {
+        "max_age": "100000"
+      },
+      "validator": {
+        "pub_key_types": [
+          "ed25519"
+        ]
+      }
+    }
+  }
+}
+```
+
+### UnconfirmedTxs
+
+Get a list of unconfirmed transactions.
+
+#### Parameters
+
+- `limit (integer)` The amount of txs to respond with.
```

Comment:
I feel like this should be paginated or something. In tendermint-go this has a upper bound of 100 i.e. you can only request the top 100 txs

Reply:
agree, it would be good to paginate more things.